packages/tx/src/message-composers/complex/swap.ts

```diff
@@ -0,0 +1,246 @@
+  coinAmount,
+  maxSlippage,
+  quote,
+  tokenInCoinMinimalDenom,
+  tokenOutCoinMinimalDenom,
+  tokenInCoinDecimals,
+  tokenOutCoinDecimals,
+  quoteType,
+}: {
+  coinAmount: string;
+  maxSlippage: string;
+  quote: QuoteOutGivenIn | QuoteInGivenOut | undefined;
+  tokenInCoinMinimalDenom: string;
+  tokenOutCoinMinimalDenom: string;
+  tokenInCoinDecimals: number;
+  tokenOutCoinDecimals: number;
+  quoteType: QuoteType;
+}) {
+  if (isNil(quote)) {
+    throw new Error(
+      "User input should be disabled if no route is found or is being generated"
+    );
+  }
+  if (isNil(coinAmount)) throw new Error("No input");
+  if (isNil(tokenInCoinMinimalDenom)) throw new Error("No from asset");
+  if (isNil(tokenOutCoinDecimals)) throw new Error("No to asset");
+
+  if (quoteType === "out-given-in") {
+    const routes: SwapTxRouteOutGivenIn[] = [];
+
+    for (const route of quote.split) {
+      const pools: SwapTxPoolOutGivenIn[] = [];
+      const typedRoute = route as RouteOutGivenIn;
+      for (let i = 0; i < route.pools.length; i++) {
+        const pool = route.pools[i];
+
+        pools.push({
+          id: pool.id,
+          tokenOutDenom: typedRoute.tokenOutDenoms[i],
+        });
+      }
+
+      routes.push({
+        pools: pools,
+        tokenInAmount: route.initialAmount.toString(),
+      });
+    }
+
+    /** In amount converted to integer (remove decimals) */
+    const tokenIn = {
+      coinMinimalDenom: tokenInCoinMinimalDenom,
+      amount: coinAmount,
+    };
+
+    /** Out amount with slippage included */
+    const tokenOutMinAmount = quote.amount
+      .toDec()
+      .mul(DecUtils.getTenExponentNInPrecisionRange(tokenOutCoinDecimals))
+      .mul(new Dec(1).sub(new Dec(maxSlippage)))
+      .truncate()
+      .toString();
+
+    return {
+      routes,
+      tokenIn,
+      tokenOutMinAmount,
+    };
+  } else {
+    const routes: SwapTxRouteInGivenOut[] = [];
+
+    for (const route of quote.split) {
+      const pools: SwapTxPoolInGivenOut[] = [];
+      const typedRoute = route as RouteInGivenOut;
+
+      for (let i = 0; i < route.pools.length; i++) {
+        const pool = route.pools[i];
+        pools.push({
+          id: pool.id,
+          tokenInDenom: typedRoute.tokenInDenoms[i],
+        });
+      }
+
+      routes.push({
+        pools: pools,
+        tokenOutAmount: route.initialAmount.toString(),
+      });
+    }
+
+    /** In amount converted to integer (remove decimals) */
+    const tokenOut = {
+      coinMinimalDenom: tokenOutCoinMinimalDenom,
+      amount: coinAmount,
+    };
+
+    /** Out amount with slippage included */
+    const tokenInMaxAmount = quote.amount
+      .toDec()
+      .mul(DecUtils.getTenExponentNInPrecisionRange(tokenInCoinDecimals))
+      .mul(new Dec(1).add(new Dec(maxSlippage)))
+      .truncate()
+      .toString();
+
+    return {
+      routes,
+      tokenOut,
+      tokenInMaxAmount,
+    };
+  }
+}
+
+export async function getSwapMessages({
```

Comment:
I think @mattupham is also using this. Good we organized/moved them here along with other composers. Cleaner!